### ISSUE TITLE
Feature request: Configure keep alive options for socket

### ISSUE THREAD

**Issue by NAJ10** [functional-streams-for-scala/fs2#1546]
Java 11 adds the ability to set keep alive options on a per socket basis (https://docs.oracle.com/en/java/javase/11/docs/api/jdk.net/jdk/net/ExtendedSocketOptions.html) which is a very useful thing to have when your code is running in a container and you are unable to alter the global configuration for keep alive and you need the keep alive packets to prevent tcp connections being blocked by firewalls in corporate environments due to inactivity on the connection. 

Please can we have the ability to set the keep alive options for sockets. The tricky bit here is that this functionality is only available on Java 11 and fs2 needs to support Java 8 for a long while yet.

**Comment by SystemFw** [comment]
@mpilquist any thoughts? The team right now has been forced to copy-paste-and-alter from fs2-io because of this

**Comment by mpilquist** [comment]
It sounds good to me. The simplest approach would be modifying `SocketGroup#{client, server, serverWithLocalAddress}` to take a `List[SocketOption[_]]` param.

**Comment by Qthunder** [comment]
> It sounds good to me. The simplest approach would be modifying `SocketGroup#{client, server, serverWithLocalAddress}` to take a `List[SocketOption[_]]` param.

It's not that simple in that `SocketOption` doesn't contain the value as well, it's a wrapper around a `String` that specifies the required value type. So what I want is something like

```
 type SocketMapping[A] = (SocketOption[A], A) . // Can be a case class instead
```

, where SocketGroup#{client, server, serverWithLocalAddress} take a List[SocketMapping[_]] param.

Does that sound reasonable?

**Comment by mpilquist** [comment]
Fixed by #1549
